modules/messages/src/lib.rs

```diff
@@ -746,23 +688,41 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
 		fn build(&self) {
-			PalletOperatingMode::<T, I>::put(&self.operating_mode);
+			PalletOperatingMode::<T, I>::put(self.operating_mode);
 			if let Some(ref owner) = self.owner {
 				PalletOwner::<T, I>::put(owner);
 			}
 		}
 	}
 
+	// TODO: Maybe useless for Darwinia.
 	impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		/// Get stored data of the outbound message with given nonce.
 		pub fn outbound_message_data(
```

Comment:
```rs
pub struct OutboundMessageDataInfo;
impl OutboundMessenger<AccountId32> for OutboundMessageDataInfo {
	fn check_lane_id(lane_id: &LaneId) -> bool {
		*lane_id == DARWINIA_CRAB_LANE
	}

	fn get_valid_message_sender(nonce: MessageNonce) -> Result<AccountId32, &'static str> {
		let data = BridgeDarwiniaMessages::outbound_message_data(DARWINIA_CRAB_LANE, nonce)
			.ok_or("Invalid outbound message data")?;
		let payload = bm_darwinia::ToDarwiniaMessagePayload::decode(&mut &data.payload[..])
			.map_err(|_| "decode message payload failed")?;
		match payload.origin {
			CallOrigin::SourceAccount(account_id) => Ok(account_id),
			_ => Err("Invalid Account Type"),
		}
	}
}
```

`outbound_message_data` is currently used in Darwinia, but we don't need it after the next runtime upgrade. So, it is safe to delete it in this repo's master branch.